subsys/net/ip/tcp2.c

```diff
@@ -984,13 +986,11 @@ int net_tcp_put(struct net_context *context)
 	NET_DBG("%s", conn ? log_strdup(tcp_conn_state(conn, NULL)) : "");
 
 	if (conn) {
-		tcp_out(conn, FIN | ACK);
```

Comment:
Without these changes I could still see retransmission of FIN ACK from peer.

Reply:
I mean we need to increase the seq number.

Reply:
@rveerama1 

I checked in the wireshark, the wireshark complains "previous segment not captured". So increasing the seqence number is probably wrong here.

We need to dig a bit. Possibly the ACK in FIN_WAIT_1 needs a sequence increment?

https://github.com/ozhuraki/zephyr/commit/a25b55d779a70a6611393380db28d7c035d5fedc